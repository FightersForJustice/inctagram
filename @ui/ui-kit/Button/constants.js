export const BUTTON_COLORS = Object.freeze({
  PRIMARY: 'Primary',
  BASIC: 'Basic',
  OUTLINED: 'Outlined',
  GHOST: 'Ghost',
})

export const BUTTON_VARIATIONS = Object.freeze({
  FULL_WIDTH: 'FullWidth',
  AUTO_HEIGHT: 'AutoHeight',
  DEFAULT: 'Default',
})
