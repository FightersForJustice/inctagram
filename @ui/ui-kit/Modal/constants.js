export const CLOSE_MODAL_TEST_ID = 'closeModal'
