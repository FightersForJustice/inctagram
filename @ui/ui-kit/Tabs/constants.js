export const TAB_COLORS = Object.freeze({
  PRIMARY: 'Primary',
  GHOST: 'Ghost',
})
export const profileSettingsData = [
  { value: 'general', label: 'General information' },
  { value: 'Devices', label: 'Devices' },
  { value: 'AccountManagment', label: 'Account Managment' },
  { value: 'Payments', label: 'MyPayments' },
]
