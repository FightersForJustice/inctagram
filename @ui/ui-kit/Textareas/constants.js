export const TEXTAEREA_COLORS = Object.freeze({
  DEFAULT: 'Default',
  ACTIVE: 'Active',
  ERROR: 'Error',
})
