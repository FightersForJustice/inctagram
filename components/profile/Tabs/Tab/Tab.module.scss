@use '/styles/library' as s;
@use 'styles/mixin' as m;

.tab {
  @include m.h3;
  color: s.$dark-100;
  font-family: Inter;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  letter-spacing: 0px;
  text-align: center;
  padding: 6 15px;
  width: 182px;
  height: 36px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: transparent;
  cursor: pointer;

  &:first-child {
    border-top-left-radius: 6px;
  }
  &:last-child {
    border-top-right-radius: 6px;
  }
  &:hover {
    color: s.$dark-100;
  }
  &:focus {
    position: relative;
    box-shadow: 0 0 0 2px black;
  }

  &[data-state='active'] {
    color: s.$primary-500;
    box-shadow: inset 0 -1px 0 0 currentColor, 0 1px 0 0 currentColor;
  }
}
