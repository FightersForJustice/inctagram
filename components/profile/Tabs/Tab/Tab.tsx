import * as Tabs from '@radix-ui/react-tabs'
import style from './Tab.module.scss'

export type TabType = {
  title: string
  value: string
}

export const Tab: React.FC<TabType> = ({ title, value }) => {
  return (
    <Tabs.Trigger className={style.tab} value={value}>
      {title}
    </Tabs.Trigger>
  )
}
