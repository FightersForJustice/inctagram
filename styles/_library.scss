//vars
//mainColor

$bgColor: #0d0d0d;
$text: #fff;

@mixin btn {
    border-radius:2px;
    background: #397DF6;
    color: #FFF;
    font-size: 16px;
    font-family: Inter;
    font-style: normal;
    font-weight: 600;
    line-height: 24px;
    cursor: pointer;
  }