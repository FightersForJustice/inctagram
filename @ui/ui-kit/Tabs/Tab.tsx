import React from 'react'
import * as Tabs from '@radix-ui/react-tabs'
import { TAB_COLORS } from '@ui/ui-kit/Tabs/constants'
import styles from '@ui/ui-kit/Tabs/Tabs.module.scss'
import classNames from 'classnames'

interface TabData {
  value: string
  label: string
  disabled?: boolean
}

interface TabProps extends TabData {
  disabled?: boolean
}

export const Tab: React.FC<TabProps> = ({ value, label, disabled }) => {
  const buttonClasses = classNames(styles.tab, {
    [styles.disabled]: disabled,
  })

  return (
    <Tabs.Trigger value={value} className={buttonClasses} disabled={disabled}>
      {label}
    </Tabs.Trigger>
  )
}
