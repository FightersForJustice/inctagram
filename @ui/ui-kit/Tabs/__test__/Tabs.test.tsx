import React from 'react'
import { render, cleanup, screen, fireEvent } from '@testing-library/react'
import * as Tabs from '@radix-ui/react-tabs'
import { Tab } from '../Tab'
import styles from '@ui/ui-kit/Tabs/Tabs.module.scss'

import { TAB_COLORS } from '../constants'

const setupTab = (props = {}) => {
  const defaultProps = {
    value: 'tab1',
    label: 'Tab 1',
    ...props,
  }

  render(
    <Tabs.Root defaultValue={defaultProps.value}>
      <Tabs.List>
        <Tab {...defaultProps} />
      </Tabs.List>
    </Tabs.Root>
  )

  return {
    tab: screen.getByText(defaultProps.label),
    ...defaultProps,
  }
}

describe('Tab', () => {
  afterEach(cleanup)

  it('renders correctly', () => {
    const { tab } = setupTab()

    expect(tab).toBeInTheDocument()
  })

  it('renders with label', () => {
    const { label } = setupTab()

    expect(screen.getByText(label)).toBeInTheDocument()
  })

  it('has default styles', () => {
    const { tab } = setupTab()

    expect(tab).toHaveClass(styles.tab)
  })

  it('applies disabled class when disabled prop is true', () => {
    const { tab } = setupTab({ disabled: true })

    expect(tab).toHaveClass(styles.disabled)
  })

  it('does not apply disabled class when disabled prop is false', () => {
    const { tab } = setupTab({ disabled: false })

    expect(tab).not.toHaveClass(styles.disabled)
  })
  it('displays the correct label', () => {
    const { label } = setupTab()

    expect(screen.getByText(label)).toBeInTheDocument()
  })
  it('does not trigger onClick event when disabled', () => {
    const onClick = jest.fn()
    const { tab } = setupTab({ disabled: true, onClick })

    fireEvent.click(tab)

    expect(onClick).not.toHaveBeenCalled()
  })
  it('becomes active when selected', () => {
    const { tab, label } = setupTab()
    const activeTab = screen.getByRole('tab', { selected: true, name: label })

    expect(activeTab).toBeInTheDocument()

    fireEvent.click(tab)

    expect(activeTab).toHaveAttribute('data-state', 'active')
  })

  it('has correct ARIA roles and attributes', () => {
    const { tab, label } = setupTab()
    const tabButton = screen.getByRole('tab', { name: label })

    expect(tab).toBe(tabButton)

    fireEvent.click(tab)

    expect(tabButton).toHaveAttribute('aria-selected', 'true')
  })
})
const colorProps = [TAB_COLORS.PRIMARY, TAB_COLORS.GHOST]
const disabledProps = [true, false]
