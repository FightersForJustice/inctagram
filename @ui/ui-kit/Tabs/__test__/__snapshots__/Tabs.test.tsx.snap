// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot correctly 	 disabled: false 	
    value: 'value' 	
    lable: 'label'
             1`] = `
<DocumentFragment>
  <div
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      data-orientation="horizontal"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-:r2:-content-value"
        aria-selected="false"
        class="tab"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        id="radix-:r2:-trigger-value"
        role="tab"
        tabindex="-1"
        type="button"
      >
        label
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`render snapshot correctly 	 disabled: true 	
    value: 'value' 	
    lable: 'label'
             1`] = `
<DocumentFragment>
  <div
    data-orientation="horizontal"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      data-orientation="horizontal"
      role="tablist"
      style="outline: none;"
      tabindex="-1"
    >
      <button
        aria-controls="radix-:r0:-content-value"
        aria-selected="false"
        class="tab disabled"
        data-disabled=""
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-state="inactive"
        disabled=""
        id="radix-:r0:-trigger-value"
        role="tab"
        tabindex="-1"
        type="button"
      >
        label
      </button>
    </div>
  </div>
</DocumentFragment>
`;
